CinemaTicketSalesSystem/CinemaTicketSalesBusinessLogic/CRUD/MovieService.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Data.Entity;
+using ApplicationDbMovies.Contexts;
+using ApplicationDbMovies.Models;
+using CinemaTicketSalesBusinessLogic.Interfaces;
+using AutoMapper;
+using CinemaTicketSalesBusinessLogic.Models;
+
+namespace CinemaTicketSalesBusinessLogic.Queries
+{
+    public class MovieService : IMovieService
+    {
+        private readonly ApplicationDbContext _db;
+        private readonly IMapper _mapper;
+
+        public MovieService(ApplicationDbContext context, IMapper mapper)
+        {
+            _db = context;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Using in HomeController for viewing Movies collection on the Main page.
+        /// </summary>        
+        public IEnumerable<MovieInfoModel> GetMovies()
+        {
+            var movies = _db.Movies.Include(x => x.Pictures).ToList();
+            var movieModels = movies.Select(m =>
+                {
+                    var movieModel = _mapper.Map<Movie, MovieInfoModel>(m);
+                    movieModel.Url = m.Pictures.FirstOrDefault()?.Url;
+                    movieModel.MovieId = m.Id;
```

Comment:
you can move it to mapping configuration
Map(d => d.MovieId, o => o.MapFrom(s => s.Id))